typescript/agentkit/src/action-providers/flaunch/utils.ts

```diff
@@ -0,0 +1,712 @@
+  PinSize: number;
+  Timestamp: string;
+  isDuplicate: boolean;
+}
+
+interface CoinMetadata {
+  name: string;
+  description: string;
+  image: string;
+  external_link: string;
+  collaborators: string[];
+  discordUrl: string;
+  twitterUrl: string;
+  telegramUrl: string;
+}
+
+interface IPFSParams {
+  metadata: {
+    base64Image: string;
+    description: string;
+    websiteUrl?: string;
+    discordUrl?: string;
+    twitterUrl?: string;
+    telegramUrl?: string;
+  };
+  pinataConfig: PinataConfig;
+}
+
+interface TokenUriParams {
+  metadata: {
+    imageUrl: string;
+    description: string;
+    websiteUrl?: string;
+    discordUrl?: string;
+    twitterUrl?: string;
+    telegramUrl?: string;
+  };
+  pinataConfig: PinataConfig;
+}
+
+/**
+ * Uploads a base64 image to IPFS using Pinata
+ *
+ * @param params - Configuration and base64 image data
+ * @param params.pinataConfig - Pinata configuration including JWT
+ * @param params.base64Image - Base64 encoded image data
+ * @param params.name - Optional name for the uploaded file
+ * @param params.metadata - Optional metadata key-value pairs
+ * @returns Upload response with CID and other details
+ */
+const uploadImageToIPFS = async (params: {
+  pinataConfig: PinataConfig;
+  base64Image: string;
+  name?: string;
+  metadata?: Record<string, string>;
+}): Promise<UploadResponse> => {
+  try {
+    const formData = new FormData();
+
+    // Convert base64 to Blob and then to File
+    // Remove data URL prefix if present (e.g., "data:image/jpeg;base64,")
+    const base64Data = params.base64Image.split(",")[1] || params.base64Image;
+    const byteCharacters = atob(base64Data);
+    const byteArrays: Uint8Array[] = [];
+
+    for (let offset = 0; offset < byteCharacters.length; offset += 1024) {
+      const slice = byteCharacters.slice(offset, offset + 1024);
+      const byteNumbers = new Array(slice.length);
+      for (let i = 0; i < slice.length; i++) {
+        byteNumbers[i] = slice.charCodeAt(i);
+      }
+      const byteArray = new Uint8Array(byteNumbers);
+      byteArrays.push(byteArray);
+    }
+
+    // Detect mime type from base64 string
+    let mimeType = "image/png"; // default
+    if (params.base64Image.startsWith("data:")) {
+      mimeType = params.base64Image.split(";")[0].split(":")[1];
+    }
+
+    const blob = new Blob(byteArrays, { type: mimeType });
+    const fileName = params.name || `image.${mimeType.split("/")[1]}`;
+    const file = new File([blob], fileName, { type: mimeType });
+
+    formData.append("file", file);
+
+    const pinataMetadata = {
+      name: params.name || null,
+      keyvalues: params.metadata || {},
+    };
+    formData.append("pinataMetadata", JSON.stringify(pinataMetadata));
+
+    const pinataOptions = {
+      cidVersion: 1,
+    };
+    formData.append("pinataOptions", JSON.stringify(pinataOptions));
+
+    const response = await axios.post("https://api.pinata.cloud/pinning/pinFileToIPFS", formData, {
```

Comment:
If possible, refactor to `fetch` rather than `axios`. We do not have a explicit dependency on `axios` yet, and would like to keep the bundle size lean when possible